### ISSUE TITLE
Distance calculation?

### ISSUE THREAD

**Issue by Sakshi-agarwal8** [d36choi/hisafety#6]
Hello,
How did you calculate the **distance** between two lines drawn using coordinates i.e. 100 mentioned in velocity.py in algorithms folder?
Please help!

**Comment by d36choi** [comment]
It’s just loosely predicted value. (not real distance)
so this project can be used very limited situation.
Im sorry I can not help.

2021년 3월 15일 (월) 23:22, Sakshi ***@***.***>님이 작성:

**Comment by Sakshi-agarwal8** [comment]
Okay. Can you help me by letting me know how did you predict it?
And how did you find the coordinates on road using which you have drawn the lines?
Also can you please tell what else is used with the self assumed values?

**Comment by d36choi** [comment]
coordinate is object's (In this case, car) rectangle's center point. 
1. Draw a line at right angles to the road.
2. if object's coordinate passes first line. It calculates frame until the object passes second line.
3. pixel distance between two line * '(  self assumed value ) m/pixel' / frame * '(self assumed value) hour/frame'
4. caculates velocity from 3.

**Comment by Sakshi-agarwal8** [comment]
I am not getting it. 
Actually I am working on a different video. So I need to draw lines on it and then by using distance between those lines, I need to calculate speed.
You have drawn lines using coordinates [480,300] [700,300] & [300,500] [800,500] and used distance as 100.
I need to know how to get this distance and coordinates to draw lines in any video?
I really need to know. Please tell in detail in person.
https://www.linkedin.com/in/sakshi-agarwal-5499a1190/

**Comment by d36choi** [comment]
It needs to re-draw lines every time when the input video changes. (by
developer's eye measure... because we don't know real distance in such
sample video)
*distance between two lines : 100*
It is just also eye measure.
If you know *real *distance measure in your video, you can change that
value (maybe).

I think my project is not good enough to fit your goal. It's very limited.


2021년 3월 16일 (화) 오후 2:51, Sakshi ***@***.***>님이 작성:

**Comment by Sakshi-agarwal8** [comment]
Yes, lines need to be drawn every time and distance to be known. 
That's what I wanted to know.
How did you draw lines using [480,300] [700,300] & [300,500] [800,500]? From where did you get these for your video.

May be same way, I'll be able to get for my video.
My goal is same as you did in this project. Just need to know how to draw lines and distance thing. Doesn't matter if real or predicted. 

Can we connect on LinkedIn. I have given link to my profile above.

**Comment by Sakshi-agarwal8** [comment]
Please answer my queries. It would be great help...

How did you get these points [480,300] [700,300] & [300,500] [800,500] for your video to draw 2 lines? And then how to select distance if not known the real distance? How did you do for your video?

May be same way, I'll be able to get for my video. 
My goal is same as you did in this project. Just need to know how to draw lines and calculate/loosely predict distance.

**Comment by d36choi** [comment]
As I said, eye measure.
you have to draw by intuition.
distance parameter also same.
thats why I say this project has many flaws.
now I work on different space.
So this project will be no more enhanced...
you need to find your own way to make better..

2021년 3월 17일 (수) 02:32, Sakshi ***@***.***>님이 작성:
